lib/puppet/provider/cert_file/posix.rb

```diff
@@ -0,0 +1,59 @@
+require 'pp'
+
+Puppet::Type.type(:cert_file).provide :posix do
+  confine feature: :posix
+
+  def exists?
+    return false unless Pathname.new(resource[:path]).exist?
+
+    debug "Checking file format #{resource[:path]}"
+    localcert_file = File.read(resource[:path])
```

Comment:
This is where it will fail if the path is a directory.

```
Errno::EISDIR: Is a directory @ io_fread - /etc
```

Reply:
Yes, but the agent continues.
 
It would log something like:
```
Info: Applying configuration version 'environment rev: 2021-04-30T08:44:13+02:00 c77a23b2 Tibor Répási: test'
Error: /Stage[main]/Roles_test::Cert/Cert_file[/tmp]: Could not evaluate: Is a directory @ io_fread - /tmp
Notice: /Stage[main]/Roles_test::Cert/File[/tmp/test.txt]/ensure: defined content as '{md5}65a8e27d8879283831b664bd8b7f0ad4' (corrective)
Info: Class[Roles_test::Cert]: Unscheduling all events on Class[Roles_test::Cert]
Notice: Applied catalog in 4.35 seconds
```

Could catch and re-throw to wrap the error, like e.g.:
```
Info: Applying configuration version 'environment rev: 2021-04-30T08:44:13+02:00 c77a23b2 Tibor Répási: test'
Error: /Stage[main]/Roles_test::Cert/Cert_file[/tmp]: Could not evaluate: failed to evaluate contents of the file expected at /tmp. Caused by: Is a directory @ io_fread - /tmp
Notice: /Stage[main]/Roles_test::Cert/File[/tmp/test.txt]/ensure: defined content as '{md5}65a8e27d8879283831b664bd8b7f0ad4' (corrective)
Info: Class[Roles_test::Cert]: Unscheduling all events on Class[Roles_test::Cert]
Notice: Applied catalog in 4.32 seconds
```

This could be a bit more expressive, but the overall behaviour wouldn't change.

Reply:
Alright, fine if Puppet continues then